tests/storage_test.go

```diff
@@ -0,0 +1,592 @@
+// +build integration
+
+package inttest
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/asecurityteam/asset-inventory-api/pkg/domain"
+	"github.com/asecurityteam/asset-inventory-api/pkg/storage"
+	"github.com/asecurityteam/logevent"
+	"github.com/asecurityteam/runhttp"
+	"github.com/asecurityteam/settings"
+	packr "github.com/gobuffalo/packr/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+// db refers to a raw Postgres, without the "storage.DB" abstraction
+var db *sql.DB
+
+// dbStorage is the struct with the functions being tested
+var dbStorage *storage.DB
+var ctx context.Context
+
+func TestMain(m *testing.M) {
+	ctx = context.Background()
+	ctx = logevent.NewContext(ctx, stdoutLogger())
+	source, err := settings.NewEnvSource(os.Environ())
+	if err != nil {
+		panic(err.Error())
+	}
+
+	postgresConfigComponent := &storage.PostgresConfigComponent{}
+	dbStorage = new(storage.DB)
+	if err = settings.NewComponent(ctx, source, postgresConfigComponent, dbStorage); err != nil {
+		panic(err.Error())
+	}
+
+	db, err = connectToDB()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestNoDBRows(t *testing.T) {
+
+	before(t, db)
+
+	// code should tolerate no data in the tables
+
+	from, _ := time.Parse(time.RFC3339, "2019-08-08T08:29:35+00:00")
+	to, _ := time.Parse(time.RFC3339, "2019-08-10T08:29:35+00:00")
+	networkChangeEvents, err := dbStorage.GetIPAddressesForTimeRange(ctx, from, to)
+	if err != nil {
+		t.Fatal(err.Error())
```

Comment:
This can be replaced with `require.NoError(t, err)`. `require` is the fatal version of `assert` where `assert` will actually continue the test but register the error.